net/peer/peer.go

```diff
@@ -320,3 +331,16 @@ func (p *peer) Close() (err error) {
 	log.Debug("peer close", zap.String("peerId", p.id))
 	return p.MultiConn.Close()
 }
+
+func (p *peer) ProvideStat() *Stat {
+	protoVersion, _ := CtxProtoVersion(p.Context())
+	p.mu.Lock()
+	subConnectionsCount := len(p.active)
+	p.mu.Unlock()
+	return &Stat{
+		PeerId:         p.id,
+		SubConnections: subConnectionsCount,
+		Created:        p.created,
```

Comment:
add a field called `AliveTimeSecs` which will tell how many seconds it is alive

Reply:
Just write pls p.Lock(), defer p.Unlock() and then return &Stat{...}

Reply:
How do you propose to calculate `AliveTimeSecs`, because don't see such field in peer and? 🧐

Reply:
Fixed